src/convert.js

```diff
@@ -15,7 +14,7 @@ const JSON_OUTPUT = `${OUTPUT_DIR}/swagger.json`
 const convertOpenApi = async function () {
   const [openapiDef] = await Promise.all([
     SwaggerParser.validate(YAML_INPUT, { dereference: { circular: false } }),
-    makeDir(OUTPUT_DIR),
+    mkdir(OUTPUT_DIR, { recursive: true }, () => {}),
```

Comment:
Shouldn't this be using the promise version of `mkdir`? https://nodejs.org/api/fs.html#fspromisesmkdirpath-options

Reply:
> Shouldn't this be using the promise version of `mkdir`? https://nodejs.org/api/fs.html#fspromisesmkdirpath-options

I agree.

What should we do when eslint shows the following error?

The 'fs/promises' is not supported until Node.js 14.0.0. The configured version range is '>=8.0.0'.eslint[n/no-unsupported-features/node-builtins](https://github.com/weiran-zsd/eslint-plugin-node/blob/HEAD/docs/rules/no-unsupported-features/node-builtins.md)


Should we update eslint or just silent the validation error?


The project currently uses 

```
"eslint": "^8.38.0",
"eslint-config-prettier": "^8.8.0",
"eslint-plugin-import": "^2.27.5",
"eslint-plugin-n": "^15.7.0",
"eslint-plugin-prettier": "^4.2.1",
```

And the `.nvmrc` indicates to use node `14`

Reply:
Good question!
We're only testing on latest and as you said the `.nvmrc` indicates 14. We should probably also make it so that the `engines` property in `package.json` indicates >=14 as well, and that'd make it so that eslint picks that version as the minimum req for that rule, according to [the docs](https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-unsupported-features/node-builtins.md#version).

Reply:
> Good question! We're only testing on latest and as you said the `.nvmrc` indicates 14. We should probably also make it so that the `engines` property in `package.json` indicates >=14 as well, and that'd make it so that eslint picks that version as the minimum req for that rule, according to [the docs](https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/no-unsupported-features/node-builtins.md#version).

So, I propose the following changes:

- rename PR as **refactor: replace promisify with fs/promises version convert.js**
- node version in `.nvmrc` as `21.5.0`
-  node engine >=14 in  `package.json`  `"engines": { "node": ">=14" }`
- and later create a PR in order to migrate `promisify` with the promise version of mkdir, writeFile throughout the project

What do you think?

Reply:
I wouldn't touch the `.nvmrc` file in this change, so we can keep it to a minimum. If we're saying the minimum supported version is 14, it's not bad to have that in the local setup. I'd keep it as-is for now, and we can look at the local version changes at some other time (maybe with testing or some other JS setup).

Other than that, I think the plan makes sense!